### PR TITLE
Question type classification

### PR DESCRIPTION
**Pull request by danyaljj**
Adding application: Question Type Classification.  Details in the readme.

### CODE REVIEW

**Comment by kordjamshidi** [comment]
It seems I do not have write access and permission to merge? @bhargav

**Comment by bhargav** [comment]
@kordjamshidi Strange. You had `Admin` permissions already. Anyways I also made the [Saul team](https://github.com/orgs/IllinoisCogComp/teams/saul) to have Admin privileges to the repository. Do you have write access/permissions now.

**Comment by kordjamshidi** [comment]
I see that I have the permissions now, thanks. This was not the case the other day.

File: saul-core/src/main/scala/edu/illinois/cs/cogcomp/saul/datamodel/DataModel.scala [hunk 1]
```diff
@@ -245,7 +245,7 @@ trait DataModel extends Logging {
     def apply(f: T => List[String])(implicit tag: ClassTag[T], d1: DummyImplicit, d2: DummyImplicit, d3: DummyImplicit,
       d4: DummyImplicit, d5: DummyImplicit, d6: DummyImplicit): DiscreteCollectionProperty[T] = {
       def cachedF = if (cache) { x: T => getOrUpdate(x, f).asInstanceOf[List[String]] } else f
-      val a = new DiscreteCollectionProperty[T](name, cachedF, ordered) with NodeProperty[T] {
+      val a = new DiscreteCollectionProperty[T](name, cachedF, !ordered) with NodeProperty[T] {
```
Comment (Rahgooy) [inline saul-core/src/main/scala/edu/illinois/cs/cogcomp/saul/datamodel/DataModel.scala]: @danyaljj why the order negated here?

Comment (danyaljj) [inline saul-core/src/main/scala/edu/illinois/cs/cogcomp/saul/datamodel/DataModel.scala]: This is a bug that existed in the system from a long time ago and I noticed it when I was making these changes (also mention in #454). 
But fixing it required retraining systems, which at the time I didn't have time to do it. So I left it as an issue.

**Pull request merged**